lib/spack/spack/spec.py

```diff
@@ -1253,8 +1172,9 @@ def from_node_dict(node):
                 if name in _valid_compiler_flags:
                     spec.compiler_flags[name] = value
                 else:
-                    spec.variants[name] = VariantSpec(name, value)
-
+                    spec.variants[name] = VariantSpec.from_node_dict(
+                        name, value
+                    )
         elif 'variants' in node:
```

Comment:
we can probably back it out in 1.0, but let's leave it in for now so that installations like EPFL can potentially read their old specs.